api/v1beta1/types.go

```diff
@@ -579,6 +579,42 @@ type SubnetSpec struct {
 	SubnetClassSpec `json:",inline"`
 }
 
+// AzureNetworkInterface defineds a network interface.
+// +optional
```

Comment:
I don't think we need an `// +optional` comment tag for a struct, only fields within a struct